common/src/IO/ExportOptions.h

```diff
@@ -0,0 +1,41 @@
+/*
+ Copyright (C) 2010-2017 Kristian Duske
+
+ This file is part of TrenchBroom.
+
+ TrenchBroom is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ TrenchBroom is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with TrenchBroom. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "IO/Path.h"
+
+namespace TrenchBroom {
+    namespace IO {
+        class ExportOptions {
+        public:
+            Path exportPath;
+        };
+
+        class MapExportOptions : public ExportOptions {
```

Comment:
I would not model this with inheritance at all. I know it's all over TB's source, but my views on this have changed. I think you should use `std::variant` to model this, i.e. `using ExportOptions = std::variant<MapExportOptions, ObjExportOptions>` and then use `std::visit` with `kdl::overload` to restore the types. Seach for `kdl::overload` in the source code to find lots of examples on how to do this.